### DISCUSSION TITLE
Sprite: Back and forth

### DISCUSSION THREAD

**Discussion by mrpapillon21** [pixijs/pixijs#8290] [Q&A]
I just want to create a box which moves all time through width limits of my background. How can I change the sprite.x direction?

I don't know, maybe something like this?
```
const
speed = 10,
box = PIXI.Sprite.from("box.png");
box.position.set(1191, 400);

app.loader
    .add('bg', "background.png")
    .load(setup);

function setup()
{
    const texture = app.loader.resources.bg.texture;

    const verticesX = 10;
    const verticesY = 10;
    const plane = new PIXI.SimplePlane(texture, verticesX, verticesY);

    plane.x = 0;
    plane.y = 0;
    app.stage.addChild(plane);

    app.stage.addChild(box);
    
    app.ticker.add(gameLoop);

function gameLoop(delta)
{
  box.x -= speed;
  if (box.x.position <= 0)
  {
    box.x += speed;
  }
}
```

Sorry for my english :/

**Comment by miltoncandelero** [comment]
if that speed is the speed for the box and only for the box, you can change the sign of the speed variable with
`speed = Math.abs(speed)` to make it positive or `speed = Math.abs(speed) * -1` to make it negative. If you do that, you only add it to the box in the update loop. The negative will do the rest.

Otherwise, keep a boolean value determining if you should add or subtract the speed and change that boolean when the condition is met.
